providers/implementations/kem/template_kem.c

```diff
@@ -142,10 +142,8 @@ static int template_encapsulate(void *vctx, unsigned char *out, size_t *outlen,
     if (secretlen != NULL)
         *secretlen = 0; /* replace with real shared secret length */
 
-    if (out == NULL) {
-        debug_print("encaps outlens set to %d and %d\n", *outlen, *secretlen);
```

Comment:
Is there no other way than to completely remove the debug output?

Reply:
I've decided to remove the `debug_print()` here is that it will always produce something like this:
```
encaps outlens set to 0 and 0
```
given I understand the logic right.

Reply:
In the template, yes. But in a real implementation those values do have values other than 0. And then the output does become _very_ relevant indeed.